src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -564,43 +536,31 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
     void ConptyConnection::Close() noexcept
     try
     {
-        const bool isClosing = _transitionToState(ConnectionState::Closing);
-        if (isClosing || _isStateAtOrBeyond(ConnectionState::Closed))
-        {
-            // EXIT POINT
-
-            // _clientExitWait holds a CreateThreadpoolWait() which holds a weak reference to "this".
-            // This manual reset() ensures we wait for it to be teared down via WaitForThreadpoolWaitCallbacks().
-            _clientExitWait.reset();
-
-            _hPC.reset(); // tear down the pseudoconsole (this is like clicking X on a console window)
-
-            // CloseHandle() on pipes blocks until any current WriteFile()/ReadFile() has returned.
-            // CancelSynchronousIo prevents us from deadlocking ourselves.
-            // At this point in Close(), _inPipe won't be used anymore since the UI parts are torn down.
-            // _outPipe is probably still stuck in ReadFile() and might currently be written to.
-            if (_hOutputThread)
-            {
-                CancelSynchronousIo(_hOutputThread.get());
-            }
-
-            _inPipe.reset(); // break the pipes
-            _outPipe.reset();
-
-            if (_hOutputThread)
-            {
-                // Waiting for the output thread to exit ensures that all pending _TerminalOutputHandlers()
-                // calls have returned and won't notify our caller (ControlCore) anymore. This ensures that
-                // we don't call a destroyed event handler asynchronously from a background thread (GH#13880).
-                LOG_LAST_ERROR_IF(WAIT_FAILED == WaitForSingleObject(_hOutputThread.get(), INFINITE));
-                _hOutputThread.reset();
-            }
+        // .reset()ing either of these two will signal ConPTY to send out a CTRL_CLOSE_EVENT to all attached clients.
+        // FYI: The other members of this class are concurrently read by the _hOutputThread
+        // thread running in the background and so they're not safe to be .reset().
+        _hPC.reset();
+        _inPipe.reset();
+
+        _transitionToState(ConnectionState::Closing);
+
+        // CloseHandle() on pipes blocks until any current WriteFile()/ReadFile() has returned.
+        // CancelSynchronousIo prevents us from deadlocking ourselves.
+        // At this point in Close(), _inPipe won't be used anymore since the UI parts are torn down.
+        // _outPipe is probably still stuck in ReadFile() and might currently be written to.
+        CancelSynchronousIo(_hOutputThread.get());
+        // Waiting for the output thread to exit ensures that all pending _TerminalOutputHandlers()
+        // calls have returned and won't notify our caller (ControlCore) anymore. This ensures that
+        // we don't call a destroyed event handler asynchronously from a background thread (GH#13880).
+        LOG_LAST_ERROR_IF(WAIT_FAILED == WaitForSingleObject(_hOutputThread.get(), INFINITE));
```

Comment:
what if there's no output thread?

Reply:
A `_hOutputThread` exists as long as `Start()` was previously called, which should be quite likely. I can add an if condition however, since it can't hurt to do so, right?